src/server/modules/post/post.router.ts

```diff
@@ -186,6 +186,26 @@ export const postRouter = router({
       })
       return updatedPost
     }),
+  search: protectedProcedure
+    .input(
+      z.object({
+        query: z.string(),
+        limit: z.number().min(1).max(30).optional().default(10),
+      })
+    )
+    .query(async ({ input: { query, limit }, ctx }) => {
+      return await ctx.prisma.post.findMany({
+        take: limit,
+        where: {
+          deletedAt: null,
+          OR: [
+            { title: { contains: query } },
+            { content: { contains: query } },
```

Comment:
@karrui example:
title: hello world
content: dog mouse cat

searching "hello world dog" will fail since neither the title nor content contains these 3 words

so something like CONCAT(title, ' ', content) ~ "%dog%mouse%cat%" OR CONCAT(content,' ', title) ~ "%dog%mouse%cat%"

Reply:
i see. seems like edge case with no real need to support (since we don't even have good full text search 😢) will ignore for now